crates/lunatic-process-api/src/lib.rs

```diff
@@ -633,21 +633,29 @@ fn link<T: ProcessState + ProcessCtx<T>>(
     let this_process = WasmProcess::new(id, signal_mailbox.0);
 
     // Send link signal to other process
-    let process = caller
-        .data()
-        .environment()
-        .get_process(process_id)
-        .or_trap("lunatic::process::link")?
-        .clone();
-    process.send(Signal::Link(tag, Arc::new(this_process)));
-
-    // Send link signal to itself
-    caller
-        .data_mut()
-        .signal_mailbox()
-        .0
-        .send(Signal::Link(tag, process))
-        .expect("The signal is sent to itself and the receiver must exist at this point");
+    let process = caller.data().environment().get_process(process_id);
+
+    if let Some(process) = process {
+        process.send(Signal::Link(tag, Arc::new(this_process)));
+
+        // Send link signal to itself
+        caller
+            .data_mut()
+            .signal_mailbox()
+            .0
+            .send(Signal::Link(tag, process))
+            .expect("The Link signal is sent to itself and the receiver must exist at this point");
+    } else {
+        // Link died (currently death reason normal is sent)
+        caller
+            .data_mut()
+            .signal_mailbox()
+            .0
+            .send(Signal::LinkDied(process_id, tag, DeathReason::Normal))
```

Comment:
I think we should use another `DeathReason` here. Erlang uses the `noproc` exit signal, that will fail the process by default if signals are not trapped. We should maybe use something similar to indicate that the action of linking has failed, as we can't really tell why the process finished (normal, failure, ...).

Note: The Erlang behaviour is actually a bit complicated. If it's a local process (on the same node) and the check of process existence is "fast", the caller raises a `noproc` exception, otherwise it receives a `noproc` signal. This is done like this because of historical reasons. We don't need to keep this inconsistent behaviour and can always send a signal. Raising an exception in the guest wouldn't even be an option for us.

Reply:
I've added `NoProcess` reason but the behaviour is the same as with normal.